src/search.ts

```diff
@@ -36,6 +37,12 @@ export interface SearchResult {
     lastResultUrl?: string;
 }
 
+export interface SearchFilter {
+    filterKey: string;
+    filterValue: string;

```

Comment:
Note: I converted this to just `string` as opposed to `string[]` since it seemed to deviate from how `ElasticSearchService.filterRulesForActiveResources` was being used previously. Happy to change this back to `string[]` if you can share some use cases that require this to be an array.

Reply:
I'm currently implementing `getSearchFilterBasedOnIdentity` in `authz-smart` package. One use case I was thinking about, is how do we want to represent an "or" operation? For example, if we want to return all FHIR resources in which there is a reference to a Patient's relative url `or` full url, how would we represent that?

Thoughts on this interface
```
export interface SearchFilter {
    key: string;
    value: string[];
    operator: '==' | '!=' | '>' | '<' | '>=' | '<=';
    logicalOperator: 'AND' | 'OR' | 'N/A';
}
```

Then to create a filter for the full url or absolute url of a Patient, we can do this.
```
{
    key: '_reference',
    value: ['Patient//1234', 'http://fhirserver.com/Patient/1234'],
    operator: '==',
    logicalOperator: 'OR',
}
```
                    ```

Reply:
Ah yep, this use case makes more sense now. The logical `AND` combinations  can also be expressed by using multiple `SearchFilters`. This particular interface though would limit us to logical `OR` combinations of the same `operator`. Is there a use case where we might have to express an `OR` combination of two different operators? And if so, would it make more sense to define another interface that defines how instances of `SearchFilter` are combined? 

A contrived example: 
`_reference == Patient/1234 OR age > 21` 

ElasticSearch uses Boolean queries as compound queries to wrap other compound or leaf queries with roughly the equivalent of logical operators:  

```
must|filter -> AND
should -> OR
must_not -> NOT
```
https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-bool-query.html

Maybe we add an additional interface like this?

```
export interface SearchFilterGroup {
    searchFilters: SearchFilter[];
    logicalOperator: 'AND' | 'OR' | 'NOT' = ;
}
```

Reply:
Great callout - we talked about this offline. We couldn't think of a use-case (yet) of having a `SearchFilterGroup`. Do you guys have any use-cases? Also bear in mind that `search` in itself does have OR built in for a key value pair itself but not across multiple pairs. Similar to our approach so far: https://www.hl7.org/fhir/search.html#combining

Reply:
We don't have a use case like this so far either. Should we skip for now?

Reply:
Yep, let's skip for now. In the future, if we find that we need to support this use case, we can add the `SearchFilterGroup` interface.

Reply:
Makes sense, I updated to add the `logicalOperator` and convert the `value` into `string[]`.